src/test/java/org/apache/log4j/LoggerTest.java

```diff
@@ -499,6 +501,30 @@ public void testAppenderEvent4() {
         assertEquals(1, listener.getRemoveEventCount());
     }
 
+    @Test
+    public void testConsoleAppender() {
+        Logger logger = Logger.getLogger("testconsole.appender");
+        ByteArrayOutputStream arrayOut = null;
+        PrintStream out = System.out;
+        try {
+            arrayOut = new ByteArrayOutputStream();
+            System.setOut(new PrintStream(arrayOut));
+            ConsoleAppender consoleAppender = new ConsoleAppender();
+            //consoleAppender.setFollow(true);
+            consoleAppender.setTarget("System.out");
+            consoleAppender.activateOptions();
+            logger.addAppender(consoleAppender);
+            consoleAppender.setLayout(new SimpleLayout());
+            logger.warn("hello console appender");
+
+            // ConsoleAppender redirects to real stdout, so nothing in the arrayOut
+            // this is failed before ConsoleAppender changes.
+            assertEquals("", arrayOut.toString());
```

Comment:
This will fail if revert the changes in ConsoleAppender.

I don't have good way to test that the log message is redirected to real <code>stdout</code>, any idea about that?